backend/data/seed.js

```diff
@@ -82,22 +92,22 @@ export async function seedDatabase() {
   } catch (error) {
     console.error('Error while seeding database:', error);
   } finally {
-    await mongoose.disconnect();
+    !memoryMongo && await mongoose.disconnect();
```

Comment:
To clarify, this line is meant to be like a compact if statement? The goal is to have mongoose disconnect only if it's something other than a MongoMemoryServer in use?

Reply:
This is called a [short-circuit evaluation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND#short-circuit_evaluation). Essentially, if the left side is false, since logical AND requires both sides to be true, it simply stops and doesn't evaluate the right side of &&.

In this case, if we already have mongod running in the background either locally or in a container, we're just using `mongoose` to help us seed the mongo database then we disconnect properly. But when it comes to using the `memoryMongo`, we don't want to close it because it's keeping the MongoMemoryServer alive. If we disconnect it the database will close and that will cause our tests that use it to have a connection error.

Reply:
Gotcha thanks for the explanation. I think it comes down to taste and what you think the code standards are, but I'm not a fan of checking a conditional like this. It makes the branching implicit, and I haven't seen this style used anywhere else in the codebase. I'd suggest doing a single-line if statement instead, like `if (!memoryMongo) await mongoose.disconnect();`. But if you feel strongly about doing it this way I won't stop you.

Reply:
I agree. It's more readable too using a single-line if statement. You use short-circuiting often in JSX. This is the only place it's used and it's not that easy to understand so I'll update it!